src/github.ts

```diff
@@ -160,3 +160,46 @@ export enum Conclusion {
     // this sets the status of the check run to "in_progress"
     InProgress = ""
 }
+
+export class GitHubRelease extends Job {
+  /**
+  * Creates a GitHub release using the provided arguments.
+  * The release body is pre-configured to be a listing of commits since the last tag.
+  *
+  * @param project - the Brigade project
+  * @param tag - release tag
+  * @param binDir - optional path to the directory holding release assets (default is empty)
+  * @param workDir - optional path to the working directory (default is "/src")
+  */
+  constructor(project: Project, tag: string, binDir = "", workDir = "/src") {
+    if (!project.secrets || !project.secrets.ghToken) {
+      throw new Error("project.secrets.ghToken undefined");
+    }
+
+    if (!project.repo || !project.repo.name) {
+      throw new Error("project.repo.name undefined")
+    }
+
+    let parts = project.repo.name.split("/", 2);
+    let org = parts[0];
+    let repo = parts[1];
+
+    super(`${org}-${repo}-release`, "brigadecore/gh-release:edge")
```

Comment:
A few questions here:

- did we ever get back to having a central repo for all our images?
- if we use the same naming scheme, `:edge` is always the newest image - are we concerned that we might update this image and not the library itself?
- do we want to maybe have a parameter so people could theoretically use this image from their own registry?

Reply:
Good questions.
1. No, we haven't.  Hadn't set it as a priority since the consumers of the kind image and this new one are in this same repo.  If this PR is merged and we now have more than one image to track, do you think we should place both in a separate repo?
2. That's a valid concern.  I can be persuaded to follow best practices and cut immutable tags.
3. My initial thinking was that since the current implementation is tightly coupled with the image contents (job tasks use/invoke `git` and `ghr` CLIs, for instance), I didn't entertain the option of allowing customization of the image name.  Though, the case for supplying the same/similar image but from a private registry is valid.  WDYT? 

Stepping back, would we want to spike on an implementation that provides the same functionality but via a js library ([octokit](https://github.com/octokit/rest.js/) perhaps?) instead of the current approach of extending the Job class and tightly-coupling the tasks to the job image?

Reply:
OH!
That could be interesting, and I'm happy to pair sometime to spike that out.

That being said, regarding 1) we should maybe just include the Dockerfile as part of this repo, WDYT?

Reply:
(It can also be part of a follow-up PR, I don't necessarily mind).